src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\OrderReturn\Repository;
+
+use OrderReturn;
+use PrestaShop\PrestaShop\Adapter\OrderReturn\Validator\OrderReturnValidator;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnException;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\ValueObject\OrderReturnId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+
+class OrderReturnRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @var OrderReturnValidator
+     */
+    private $orderReturnValidator;
+
+    /**
+     * @param OrderReturnValidator $orderReturnValidator
+     */
+    public function __construct(OrderReturnValidator $orderReturnValidator)
+    {
+        $this->orderReturnValidator = $orderReturnValidator;
+    }
+
+    /**
+     * Gets legacy OrderReturn
+     *
+     * @param OrderReturnId $orderReturnId
+     *
+     * @return OrderReturn
+     *
+     * @throws OrderReturnException
+     */
+    public function get(OrderReturnId $orderReturnId): OrderReturn
+    {
+        /* @var OrderReturn $orderReturn */
+        $orderReturn = $this->getObjectModel($orderReturnId->getValue(), OrderReturn::class, OrderReturnNotFoundException::class);
+
+        return $orderReturn;
+    }
+
+    /**
+     * @param OrderReturn $orderReturn
+     *
+     * @throws CoreException
+     */
+    public function update(OrderReturn $orderReturn): void
+    {
+        $this->orderReturnValidator->validate($orderReturn);
```

Comment:
I'm not sure the update should validate.

Reply:
Well i checked some other repositories and they all do validate so I don't wanna break from current practice.
```
PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository
PrestaShop\PrestaShop\Adapter\Customer\Repository\CustomerRepository
PrestaShop\PrestaShop\Adapter\Product\Combination\Repository\CombinationRepository
```

Reply:
I don't think this is a good idea, because that is not the purpose of a repository. The `update` method should IMO act like a bridge between the object and the database.

If your repository can validate data, then it can also dispatch hooks, etc.. Then it is no more a repository, it's a manager.

Reply:
I see your point. Will change it.

Reply:
Fabien is right about the principle, but in this context we put the validation inside the repositories on purpose. The reason is simple - those validators actually just wraps up the legacy obj model validation which will still happen inside the obj model later. So its not like we could choose to validate the object or not... By calling validators in repository, we just force the wrapping of legacy errors into new core exceptions with clear error codes and don't leave the opportunity for legacy exceptions to bubble up.